packages/cardpay-sdk/sdk/reward-pool/base.ts

```diff
@@ -327,41 +264,55 @@ export default class RewardPool {
       return await waitForSubgraphIndexWithTxnReceipt(this.layer2Web3, txnHash);
     }
     let safeAddress = safeAddressOrTxnHash;
+    if (!proofArray) {
+      //proofArray can be empty e.g reward only for single rewardee
+      throw new Error('proof must be provided');
+    }
+    if (!leaf) {
+      throw new Error('leaf must be provided');
+    }
+
+    let { rewardProgramId, payee, token, amount }: FullLeaf = this.decodeLeaf(leaf) as FullLeaf;
+
     if (!rewardProgramId) {
       throw new Error('rewardProgramId must be provided');
     }
-    if (!tokenAddress) {
-      throw new Error('tokenAddress must be provided');
+    if (!leaf) {
+      throw new Error('leaf must be provided');
     }
-    if (!proof) {
-      throw new Error('proof must be provided');
+    if (!proofArray) {
+      throw new Error('proofArray must be provided');
+    }
+    if (!token) {
+      throw new Error('token must be provided');
+    }
+    if (!amount) {
```

Comment:
Amount has been removed now

Reply:
I think this is relevant for checking how much reward is being taken out and comparing that to whats avaialbe in reward pool/ whats gas costing

Reply:
Sorry, I missed the section taking the amount from the leaf and thought it was left over from when the amount needed to be specified by the user.